app/models/operatingsystem.rb

```diff
@@ -170,7 +170,11 @@ def self.find_by_to_label(str)
     os = find_by_description(str.to_s)
     return os if os
     a = str.split(" ")
-    b = a[1].split('.') if a[1]
+    if a[0] =~ /ubuntu/i
```

Comment:
`=~` has some changes in Ruby 3.2 and a better alternative is to use `.match?` when you can. Also has a better performance, so newer versions of RuboCop also flag this
```suggestion
    if a[0]&.match?(/ubuntu/i)
```

Reply:
I switched to using the normalized OS name as proposed below.